lib/magma/actions/model_update_actions.rb

```diff
@@ -0,0 +1,36 @@
+require 'magma/actions/update_attribute'
+
+class Magma
+  class ModelUpdateActions
+    def self.build(project_name, actions_list)
+      new(project_name, actions_list) 
+    end
+
+    def perform
+      return false unless valid?
+      @actions.all?(&:perform)
+    end
+
+    def errors
+      @errors + @actions.flat_map(&:errors)
+    end
+
+    private
+
+    def valid?
+      @errors.empty? && @actions.all?(&:validate)
+    end
+
+    def initialize(project_name, actions_list)
+      @errors = []
+      @actions = []
+      actions_list.each do |action_params|
+        action_class = "Magma::#{action_params[:action_name].classify}Action".constantize
+        @actions << action_class.new(project_name, action_params)
+      rescue NameError => e
+        @errors << ActionError.new(message: "Invalid action type", source: action_params[:action_name]) 
```

Comment:
This NameError presumably results from the call to #constantize - but since we also call action_class.new() it might plausibly come from there, and incorrectly rescued here. We can avoid this if we determine invalid syntax by inspection rather than exception - in this case we can easily make an explicit list of valid action names to check against before we make the constant. This logic could all be ported to Magma::Action, which itself might raise an ActionError to be caught here and enqueued.

Reply:
I went with a little bit of a compromise in https://github.com/mountetna/magma/pull/173/commits/d979ca9b5cc5480db49b1d50fa988c214beb14e0. Instead of building a list of valid action names, I used [`#safe_constantize`](https://api.rubyonrails.org/classes/String.html#method-i-safe_constantize) so it doesn't raise an error. Then it gets easier to make sure `@errors` is only collecting invalid action types.